Vagrantfile

```diff
@@ -9,14 +9,32 @@ if not File.file?(config_file)
 end
 pubstack_config = YAML::load_file(config_file)
 
-# Validate sites' docroots:
-if pubstack_config.include? 'sites'
+# If the synced folder option is specified, fix the documentroot option
+# for every site.
+if pubstack_config.include? 'synced_folder'
+  # It doesn't make sense to specify both synced_folder and shared_folders,
+  # and the path to the documentroot in the VM will break, so just bail
+  # if both are specified for whatever reason.
+  if pubstack_config.include? 'shared_folders'
+    raise Vagrant::Errors::VagrantError.new, 'You may only use the synced_folder option if shared_folders is not specified. Please fix your config.yml and try again.'
+  end
+
   pubstack_config['sites'].each {|site|
+    # Since we know the host path and the VM path in this specific case,
+    # we can do documentroot validation.
     documentroot = pubstack_config['synced_folder'] + '/' + site['vhost']['documentroot']
-    if not File.directory?(File.expand_path(documentroot))
+    if not File.directory? (File.expand_path(documentroot))
       raise Vagrant::Errors::VagrantError.new, 'The docroot ' + documentroot + ' does not exist.'
     end
+
+    site['vhost']['documentroot'] = '/var/www/html/' + site['vhost']['documentroot']
   }
+  pubstack_config['shared_folders'] = [ { "map" => pubstack_config['synced_folder'], "to" => '/var/www/html'} ]
+end
+
+# Ensure that the shared_folders configuration exists.
+if not pubstack_config.include? 'shared_folders'
```

Comment:
The idea is that under the hood, `synced_folder` is just adding an entry to `shared_folders`. If you have both specified, you get an error. If you have neither specified, you get an error. You have to use one or the other.